### ISSUE TITLE
Switch to ProductDetails

### ISSUE THREAD

**Issue by ciriousjoker** [chargebee/chargebee-android#61]
Check the warning at the top of this page:
https://developer.android.com/reference/com/android/billingclient/api/SkuDetails

**Reasons for switching:**

- Don't rely on deprecated functionality
- Is probably a requirement to support the new subscription types in the Play Console (don't quote me on this, just an assumption)
- Much more information available.
  - Specifically, billing period and billing period unit via [PricingPhase](https://developer.android.com/reference/com/android/billingclient/api/ProductDetails.PricingPhase), which would also solve the missing duration in this issue: https://github.com/chargebee/chargebee-flutter/issues/39

**Comment by cb-amutha** [comment]
@ciriousjoker Yes agree with your points. we are working on upgrading our SDK with billing library 5.0. This will be available in the end of Q3'23. Will keep update on this. Thanks.
